### PR TITLE
Reassessed tutorial accuracies

### PR DESCRIPTION
**Pull request by abhibhav14**
Fix for #176

### CODE REVIEW

**Comment by npapernot** [comment]
Thanks @abhibhav14, I'll run the Travis tests a couple more times to make sure they are not flaky anymore.

We might want to reassess the JSMA tutorial tests, I think that the threshold on the accuracy on adversarial examples is too low now that the optimizer yields more accurate models.

**Comment by abhibhav14** [comment]
@npapernot I have commented the same thing on the Issue thread, but please note that the Blackbox tutorial is still broken. The tests pass because of the reduced input and epochs and data size, but when the blackbox is trained properly, the the adversarial attack accuracy is still as high as 80%.

**Comment by npapernot** [comment]
I'll take a look at the tutorial once we've merge the `generate_np` method and the new attack 

Could we also add numpy and tensorflow seeds to the tests, to minimize the flakyness of tests?

**Comment by abhibhav14** [comment]
I added the random seeds and also changed the accuracy to that on the training samples for the two FGSM attacks. I will soon add the same for the JSMA tutorial too.

**Comment by goodfeli** [comment]
OK, thanks. Now that it's seeded, could you post the results of 3-5 runs for us to see what it does on your machine?

**Comment by abhibhav14** [comment]
I unfortunately do not get determinism, but the differences are small enough to not matter.
For the keras tutorial:
0.973, 0.9736, 0.9732, 0.9736
0.0096, 0.0106, 0.0110, 0.009
0.9496, 0.9512, 0.9498, 0.951
0.4308, 0.4288, 0.4270, 0.432
where the four rows are four runs of clean_trian_clean eval, clean_train_adv_eval, adv_train_clean_eval and adv_train_adv eval

Similar results for the pure tf tutorial are:
0.9618, 0.9642, 0.9602, 0.9638
0.0220, 0.0202, 0.0220, 0.1980
0.9410, 0.9410, 0.9436, 0.9406
0.4052, 0.4296, 0.4422, 0.4224

**Comment by goodfeli** [comment]
Seems like something is still not seeded right.
- Is it deterministic if you turn off dropout?
- Is the loss right after initialization deterministic? Or do you need to do some learning before it becomes non-deterministic?

What are the accuracies on the smaller versions of the tutorials used in the tests?

**Comment by abhibhav14** [comment]
The above results are on the test versions of the tutorials, and the results on the actual tutorials are quite similar, except the results of the adversarial training, which are much better. However, there is slightly more variation, despite the seeding.
As for the second point, there is a small difference in both the loss after initialization for the first batch, and also the prediction on the first sample before training, however this difference is beyond the third decimal place.

**Comment by goodfeli** [comment]
OK. It's a bit weird that it's still stochastic, but this is definitely an improvement.
